src/property.c

```diff
@@ -1294,122 +2337,250 @@ reflist_find(dbref obj, const char *propname, dbref tofind)
 
     ptr = get_property(obj, propname);
     if (ptr) {
-	const char *pat = NULL;
+        const char *pat = NULL;
 
 #ifdef DISKBASE
-	propfetch(obj, ptr);
+        propfetch(obj, ptr);
 #endif
-	switch (PropType(ptr)) {
-	case PROP_STRTYP:
-	    temp = PropDataStr(ptr);
-	    snprintf(buf, sizeof(buf), "%d", tofind);
-	    while (*temp) {
-		if (*temp == NUMBER_TOKEN) {
-		    pat = buf;
-		    count++;
-		} else if (pat) {
-		    if (!*pat) {
-			if (!*temp || *temp == ' ') {
-			    break;
-			}
-			pat = NULL;
-		    } else if (*pat != *temp) {
-			pat = NULL;
-		    } else {
-			pat++;
-		    }
-		}
-		temp++;
-	    }
-	    if (pat && !*pat) {
-		pos = count;
-	    }
-	    break;
-	case PROP_REFTYP:
-	    if (PropDataRef(ptr) == tofind)
-		pos = 1;
-	    break;
-	default:
-	    break;
-	}
+        switch (PropType(ptr)) {
+            /* This is copy/pasta'd in reflist_add and reflist_del */
+            case PROP_STRTYP:
+                temp = PropDataStr(ptr);
+                snprintf(buf, sizeof(buf), "%d", tofind);
+
+                while (*temp) {
+                    if (*temp == NUMBER_TOKEN) {
+                        pat = buf; /* We found a number token -- let's see
+                                    * if it is the number we want.
+                                    *
+                                    * Remember 'buf' has our ref in string
+                                    * format.
+                                    */
+                        count++;
+                    } else if (pat) {
+                        /* If 'pat' is not NULL, then we are iterating
+                         * over 'pat' and 'temp in tandem, comparing along
+                         * the way to see if the numbers match.
+                         */
+                        if (!*pat) {
+                            /* If we have reached the end of 'pat', and
+                             * we have either reached the end of temp
+                             * or we have reached a space, then we have
+                             * found our number.
+                             */
+                            if (!*temp || *temp == ' ') {
+                                break;
+                            }
+
+                            /* Otherwise, this isn't our number */
+                            pat = NULL;
+                        } else if (*pat != *temp) {
+                            /* Pat and temp are no longer in sync -- this
+                             * isn't our number.
```

Comment:
No */ here.  Intentional?  Seems unintentional...

Reply:
Yeah that was a mistake, I'll fix it.

Reply:
Reguarding the /** comments

Those are called docblocks and that allows the auto-generation of API documentation.  For instance, doxygen will scan the files, find /** comments, and turn them into API docs.  That's also what all the stuff like @ param is -- doxygen picks those up and handles them special.

I'll remove the license stuff too.

Reply:
> Yeah that was a mistake, I'll fix it.

Okidoki! :D

> Reguarding the /** comments
> 
> Those are called docblocks and that allows the auto-generation of API documentation. For instance, doxygen will scan the files, find /** comments, and turn them into API docs. That's also what all the stuff like @ param is -- doxygen picks those up and handles them special.
> 
> I'll remove the license stuff too.

Thank you for explaining!  That's neat.

Reply:
My next commit will have the doxygen build files, and when I'm done with the documentation project, I'll update Makefile to have a target for building docs (Though it will require you to install doxygen)

Reply:
This should fix all the issues you found, adds doxygen build files, and adds fbstrings as well.  Give a look when you've got a sec.  I eyeballed the tests, looks like no warnings ... fingers crossed, this can be merged.